src/main/kotlin/com/exactpro/th2/common/Module.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common
+
+import com.exactpro.th2.common.schema.configuration.Configuration
+
+/**
+ * Marker interface for module
+ */
+interface Module {
+    fun <C : Configuration> loadConfiguration(configurationProvider: ConfigurationProvider, clazz: Class<C>): C
```

Comment:
Why module needs a method to load configuration? As we discussed the configuration should be                  encapsulated into the module and it should provide the required information if necessary